src/server/apsystem.ts

```diff
@@ -230,37 +267,48 @@ export default class ActivityPubSystem {
 
   async mentionToActor (mention: string): Promise<string> {
     const { username, domain } = parseMention(mention)
-    const acct = `acct:${username}@${domain}`
-    // TODO: dynamically determine the parameter name from the host-meta file
-    const mentionURL = `https://${domain}/.well-known/webfinger?resource=${acct}`
+    let webfingerURL = `https://${domain}/.well-known/webfinger?resource=acct:${username}@${domain}`
 
-    const response = await this.fetch(mentionURL)
+    let response = await this.fetch(webfingerURL)
 
-    // throq if response not ok or if inbox isn't a string
-    if (!response.ok) {
-      throw new Error(`Cannot fetch webmention data from ${mentionURL}: http status ${response.status} - ${await response.text()}`)
+    if (!response.ok && response.status === 404) {
+      const hostMetaURL = `https://${domain}/.well-known/host-meta`
+      const hostMetaResponse = await this.signedFetch(this.publicURL, {
+        url: hostMetaURL,
+        method: 'GET',
+        headers: {}
+      })
+
+      if (!hostMetaResponse.ok) {
+        throw new Error(`Cannot fetch host-meta data from ${hostMetaURL}: http status ${hostMetaResponse.status}`)
+      }
+
+      const hostMeta: HostMeta = await hostMetaResponse.json()
```

Comment:
Host-meta isn't JSON, we need to add an XML parser.

Try this library: https://github.com/NaturalIntelligence/fast-xml-parser/blob/HEAD/docs/v4/2.XMLparseOptions.md

Here's an example XML doc: https://github.com/RangerMauve/staticpub.mauve.moe/blob/default/.well-known/host-meta